src/main/java/org/eln2/mc/common/parts/foundation/CellPart.kt

```diff
@@ -0,0 +1,181 @@
+package org.eln2.mc.common.parts.foundation
+
+import mcp.mobius.waila.api.IPluginConfig
+import net.minecraft.nbt.CompoundTag
+import net.minecraft.resources.ResourceLocation
+import net.minecraft.server.level.ServerLevel
+import org.eln2.mc.Eln2
+import org.eln2.mc.annotations.ServerOnly
+import org.eln2.mc.common.cells.foundation.CellBase
+import org.eln2.mc.common.cells.foundation.CellGraphManager
+import org.eln2.mc.common.cells.foundation.CellPos
+import org.eln2.mc.common.cells.foundation.CellProvider
+import org.eln2.mc.common.space.RelativeRotationDirection
+import org.eln2.mc.extensions.NbtExtensions.useSubTagIfPreset
+import org.eln2.mc.integration.waila.IWailaProvider
+import org.eln2.mc.integration.waila.TooltipBuilder
+import java.util.*
+
+/**
+ * This part represents a simulation object. It can become part of a cell network.
+ * */
+abstract class CellPart(
+    id: ResourceLocation,
+    placementContext: PartPlacementContext,
+    final override val provider: CellProvider
+) :
+
+    Part(id, placementContext),
+    IPartCellContainer,
+    IWailaProvider {
+
+    companion object {
+        private const val GRAPH_ID = "GraphID"
+        private const val CUSTOM_SIMULATION_DATA = "SimulationData"
+    }
+
+    /**
+     * The actual cell contained within this part.
+     * It only exists on the server (it is a simulation-only item)
+     * */
+    @ServerOnly
+    final override lateinit var cell: CellBase
+
+    final override val hasCell: Boolean
+        get() = this::cell.isInitialized
+
+    private val cellPos = CellPos(placementContext.pos, placementContext.face)
+
+    /**
+     * Used by the loading procedures.
+     * */
+    @ServerOnly
+    private lateinit var loadGraphId: UUID
+
+    @ServerOnly
+    private var customSimulationData: CompoundTag? = null
+
+    protected var isAlive = false
+        private set
```

Comment:
I implemented periodic temperature updates by having an update function enqueue itself to run after a few ticks using the `EventScheduler`. Of course, the part could be broken while an update is in the queue. `isAlive` is used to prevent working with dead parts. Another option to consider would be a "work handle", which is returned when you enqueue something using the `EventScheduler`. It would be used to cancel the work.

Reply:
Also, it may be worthwhile to move this to `Part`.